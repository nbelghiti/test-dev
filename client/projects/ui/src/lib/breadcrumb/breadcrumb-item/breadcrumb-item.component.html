<div uiFlyout *ngIf="items && items.length > 1; else asLink">
	<button
		uiFlyoutToggle
		class="c-breadcrumb-item"
		[ngClass]="{
			'c-breadcrumb-item--first': first
		}"
		[title]="value"
	>
		<span class="c-breadcrumb-item__label">
			{{label}}
		</span>

		<div class="c-breadcrumb-item__item">
			<span
				*ngIf="conflict"
				class="c-breadcrumb-item__icon"
			>
				<ui-icon
					[name]="IconName.exclamationCircle"
					[size]="IconSize.tiny"
					[type]="IconType.error"
				>
				</ui-icon>
			</span>

			<span class="c-breadcrumb-item__value">
				{{value}}
			</span>

			<span class="c-breadcrumb-item__arrow">
				<ui-icon
					[name]="IconName.caretDown"
					[size]="IconSize.small"
				>
				</ui-icon>
			</span>
		</div>
	</button>

	<ui-filter-menu
		uiFlyoutContent
		class="c-flyout__content--left"
		label="value"
		[items]="items"
		[formControl]="activeBreadcrumb"
	>
	</ui-filter-menu>
</div>

<ng-template #asLink>
	<a
		class="c-breadcrumb-item"
		[ngClass]="{
			'c-breadcrumb-item--first': first
		}"
		[routerLink]="link"
		[queryParams]="queryParams"
		[title]="value"
	>
		<span class="c-breadcrumb-item__label">
			{{label}}
		</span>

		<div class="c-breadcrumb-item__item">
			<span
				*ngIf="conflict"
				class="c-breadcrumb-item__icon"
			>
				<ui-icon
					[name]="IconName.exclamationCircle"
					[size]="IconSize.tiny"
					[type]="IconType.error"
				>
				</ui-icon>
			</span>

			<span class="c-breadcrumb-item__value">
				{{value}}
			</span>
		</div>
	</a>
</ng-template>
