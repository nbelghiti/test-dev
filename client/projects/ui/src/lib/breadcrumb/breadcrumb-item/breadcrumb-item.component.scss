// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Breadcrumb item
// =============================================================================
.c-breadcrumb-item {
	position: relative;

	display: inline-flex;
	flex-direction: column;
	padding: 1.4rem 2rem;

	text-decoration: none;

	border-right: $border-width-base solid $border-color-grey-86;
	color: $text-color-grey-46;

	&:hover {
		color: $text-color-base;

		cursor: pointer;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-breadcrumb-item__item {
	display: inline-flex;
	align-items: center;

	margin-top: 0.2rem;
}

.c-breadcrumb-item__label {
	@include extend-text-overflow();

	display: inline-block;
	max-width: 20rem;

	font-size: $font-size-2x-small;
	line-height: $line-height-normalize;
	text-align: left;
	text-transform: uppercase;

	white-space: nowrap;
}

.c-breadcrumb-item__value {
	@include extend-text-overflow();

	display: inline-block;
	max-width: 20rem;

	font-size: $font-size-1x-small;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;

	white-space: nowrap;
}

.c-breadcrumb-item__arrow {
	display: inline-block;
	margin-left: 1rem;
}

.c-breadcrumb-item__icon {
	display: inline-block;
	margin-right: 0.4rem;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-breadcrumb-item--first {
	background-color: $background-color-grey-96;
}
