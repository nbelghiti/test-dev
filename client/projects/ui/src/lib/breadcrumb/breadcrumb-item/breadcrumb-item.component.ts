import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
import { FormControl } from '@angular/forms';

import { IFilterMenuItem } from '../../filter/filter-menu/filter-menu.component.types';
import { IconName, IconSize, IconType } from '../../icon/icon.types';

@Component({
	selector: 'ui-breadcrumb-item',
	templateUrl: './breadcrumb-item.component.html',
	styleUrls: ['./breadcrumb-item.component.scss']
})
export class BreadcrumbItemComponent implements OnChanges {
	@Input() public label: string = '';
	@Input() public value: string = '';
	@Input() public items: IFilterMenuItem[] = null;
	@Input() public first?: boolean = false;
	@Input() public link?: string[];
	@Input() public conflict?: boolean = false;
	@Input() public queryParams?: { [key: string]: string };

	public IconName: typeof IconName = IconName;
	public IconSize: typeof IconSize = IconSize;
	public IconType: typeof IconType = IconType;
	public activeBreadcrumb: FormControl = new FormControl();

	public ngOnChanges(changes: SimpleChanges): void {
		if (changes.value && changes.value.currentValue) {
			this.activeBreadcrumb.setValue(
				{ value: changes.value.currentValue },
				{ emitEvent: false }
			);
		}
	}
}
