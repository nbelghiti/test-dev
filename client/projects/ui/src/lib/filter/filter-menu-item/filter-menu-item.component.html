<a
	[routerLink]="link"
	*ngIf="link; else buttonTpl"
	class="c-filter-menu-item"
	[ngClass]="{
		'c-filter-menu-item--active': active
	}"
	[title]="label"
	[queryParams]="queryParams"
>
	<span class="c-filter-menu-item__value">{{ label }}</span>

	<div class="c-filter-menu-item__icon">
		<ui-icon name="check"></ui-icon>
	</div>
</a>


<ng-template #buttonTpl>
	<button
		(click)="clickHandler()"
		class="c-filter-menu-item"
		[ngClass]="{
			'c-filter-menu-item--active': active
		}"
		[title]="label"
	>
		<span class="c-filter-menu-item__value">{{ label }}</span>

		<div class="c-filter-menu-item__icon">
			<ui-icon name="check"></ui-icon>
		</div>
	</button>
</ng-template>
