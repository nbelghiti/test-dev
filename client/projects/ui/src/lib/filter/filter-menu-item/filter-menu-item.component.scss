// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: List item
// =============================================================================
.c-filter-menu-item {
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	padding: 0.4rem 1.5rem;

	text-decoration: none;

	background: none;
	border: none;

	transition: $animation-base;

	cursor: pointer;

	&:hover {
		background-color: $ui-color-success-light;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-filter-menu-item__value {
	overflow: hidden;
	margin-right: 1.6rem;

	font-family: $font-family-secondary;
	font-size: $font-size-base;
	line-height: $line-height-base;
	text-align: left;
	text-overflow: ellipsis;
	white-space: nowrap;

	color: $text-color-base;
}

.c-filter-menu-item__icon {
	color: $ui-color-success-dark;
	opacity: 0;

	transition: opacity 0.2s ease-in-out;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-filter-menu-item--active {
	font-weight: $font-weight-ultrabold;

	background-color: $ui-color-success-light;

	.c-filter-menu-item__icon {
		opacity: 1;
	}
}
