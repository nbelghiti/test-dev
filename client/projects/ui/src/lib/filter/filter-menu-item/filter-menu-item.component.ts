import { Component, Input } from '@angular/core';

@Component({
	selector: 'ui-filter-menu-item',
	templateUrl: './filter-menu-item.component.html',
	styleUrls: ['./filter-menu-item.component.scss']
})

export class FilterMenuItemComponent {
	@Input() public label: string = 'Label';
	@Input() public active: boolean;
	@Input() public link: any;
	@Input() public	queryParams: { [key: string]: string };
	@Input() public clickHandler: () => void = () => {};
}
