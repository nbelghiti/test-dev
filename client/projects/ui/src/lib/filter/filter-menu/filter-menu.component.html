<ul class="c-filter-menu">
	<li *ngFor="let item of items" class="c-filter-menu__item">
		<ui-filter-menu-item
			[active]="item?.active"
			[label]="item[label]"
			[link]="item?.link"
			[queryParams]="item?.queryParams"
			(click)="select(item)"
		></ui-filter-menu-item>
	</li>
</ul>
