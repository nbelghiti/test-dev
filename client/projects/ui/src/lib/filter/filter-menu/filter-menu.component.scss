// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Filter menu
// =============================================================================
.c-filter-menu {
	@include extend-list-clear();

	width: 100%;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-filter-menu__item {
	margin: 0.4rem 0;
}
