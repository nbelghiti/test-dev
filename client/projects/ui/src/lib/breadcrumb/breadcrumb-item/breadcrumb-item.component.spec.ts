import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';

import { FilterMenuItemComponent } from '../../filter/filter-menu-item/filter-menu-item.component';
import { FilterMenuComponent } from '../../filter/filter-menu/filter-menu.component';
import { IconComponent } from '../../icon/icon.component';

import { BreadcrumbItemComponent } from './breadcrumb-item.component';

describe('BreadcrumbItemComponent', () => {
	let component: BreadcrumbItemComponent;
	let fixture: ComponentFixture<BreadcrumbItemComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				ReactiveFormsModule,
			],
			declarations: [
				BreadcrumbItemComponent,
				IconComponent,
				FilterMenuComponent,
				FilterMenuItemComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(BreadcrumbItemComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

