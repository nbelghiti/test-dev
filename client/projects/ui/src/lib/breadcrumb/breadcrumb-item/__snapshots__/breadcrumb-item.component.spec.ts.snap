// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItemComponent Should render correctly 1`] = `
<ui-breadcrumb-item
  IconName={[Function Object]}
  IconSize={[Function Object]}
  IconType={[Function Object]}
  activeBreadcrumb={[Function FormControl]}
  conflict="false"
  first="false"
  items="null"
  label=""
  value=""
>
  <a
    class="c-breadcrumb-item"
    href="/"
    ng-reflect-klass="c-breadcrumb-item"
    ng-reflect-ng-class="[object Object]"
    title=""
  >
    <span
      class="c-breadcrumb-item__label"
    >
        
    </span>
    <div
      class="c-breadcrumb-item__item"
    >
      
      <span
        class="c-breadcrumb-item__value"
      >
          
      </span>
    </div>
  </a>
</ui-breadcrumb-item>
`;
